src/qt/macdockiconhandler.mm

```diff
@@ -34,74 +28,13 @@ void setupDockClickHandler() {
         id delegate = objc_msgSend(appInst, sel_registerName("delegate"));
         Class delClass = (Class)objc_msgSend(delegate,  sel_registerName("class"));
         SEL shouldHandle = sel_registerName("applicationShouldHandleReopen:hasVisibleWindows:");
-        if (class_getInstanceMethod(delClass, shouldHandle))
-            class_replaceMethod(delClass, shouldHandle, (IMP)dockClickHandler, "B@:");
-        else
-            class_addMethod(delClass, shouldHandle, (IMP)dockClickHandler,"B@:");
+        class_replaceMethod(delClass, shouldHandle, (IMP)dockClickHandler, "B@:");
```

Comment:
Why did/do we need this `class_replaceMethod` stuff?

Reply:
This implements the method of the application delegate which handles the `reopen application` event (i.e., the user clicks application icon in the Dock).
Ref: [How Cocoa Applications Handle Apple Events](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ScriptableCocoaApplications/SApps_handle_AEs/SAppsHandleAEs.html)